src/Microsoft.DotNet.ProjectModel/ProjectContextCache.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    public class ProjectContextCache
+    {
+        private readonly LockFileReaderCache _lockFileReaderCache;
+
+        public ProjectContextCache(LockFileReaderCache lockFileReaderCache)
+        {
+            _lockFileReaderCache = lockFileReaderCache;
+        }
+
+        private readonly Dictionary<Tuple<string, string, string>, ProjectContext> _cache = new Dictionary<Tuple<string, string, string>, ProjectContext>();
+
+        public ProjectContext Create(string projectPath, NuGetFramework framework, IEnumerable<string> runtimeIdentifiers = null)
+        {
+            runtimeIdentifiers = runtimeIdentifiers ?? Enumerable.Empty<string>();
+            var key = Tuple.Create(projectPath, framework.ToString(), string.Join(",", runtimeIdentifiers));
```

Comment:
Can you create a custom Key type instead of using a `Tuple`? Cleaner than relying on Tuple's comparison semantics.